code/game/machinery/vending_types.dm

```diff
@@ -94,6 +94,119 @@
 	)
 	product_ads = "Only the finest!;Have some tools.;The most robust equipment.;The finest gear in space!"
 
+/obj/machinery/vending/assist/ninja
```

Comment:
Less than optimal implementation of the way that getting gear is handled. Printer added framework, even if broken since everyone can access the ninja section, for certain antags to have access to sections of the uplink to buy items with telecrystals. Using the uplink would be far more simple and intuitive than vending machines.

Reply:
Vending machines are better given how I have 100% control over what each ninja can spawn with. As you said, the uplink is currently glitched so vending machines are the best solution as of now.